src/modules/flow/random/random.c

```diff
@@ -54,19 +55,17 @@ getrandom_shim(void *buf, size_t buflen, unsigned int flags)
     int fd;
     ssize_t ret;
 
-#ifdef SYS_getrandom
+#ifdef HAVE_GETRANDOM
     /* No wrappers are commonly available for this system call yet, so
      * use syscall(2) directly. */
     long gr_ret = syscall(SYS_getrandom, buf, buflen, flags);
-    if (gr_ret >= 0)
-        return gr_ret;
+    if (gr_ret > 0 && (size_t)(unsigned long)gr_ret == buflen)
+        return 0;
```

Comment:
This is wrong: it should return `gr_ret` (the number of bytes read). Or else the check in the call site will fail